pkg/server/bundle/datastore/wrapper.go

```diff
@@ -0,0 +1,45 @@
+package datastore
+
+import (
+	"context"
+	"crypto"
+	"time"
+
+	"github.com/spiffe/spire/pkg/server/bundle/pubmanager"
+	"github.com/spiffe/spire/pkg/server/datastore"
+	"github.com/spiffe/spire/proto/spire/common"
+)
+
+// WithBundlePublisher wraps a datastore interface and provides updates to
+// bundle publishers in operations that modify the local bundle.
+func WithBundlePublisher(ds datastore.DataStore, pubManager pubmanager.PubManager) datastore.DataStore {
```

Comment:
nit: instead of passing the whole PubManager interface, this could just be a function:
```suggestion
func WithBundleUpdateCallback(ds datastore.DataStore, bundleUpdated func()) datastore.DataStore {
```

Reply:
Thank you @azdagron for this suggestion to avoid the need of the Init func! It's a lot more clear and less error prone.